Feature/debugpy

mbertoldi opened the pull request:
### **User description**
Added debugpy support for debugpy (https://github.com/microsoft/debugpy).
debugpy dependency added to developer profile.
gnr web serve now has a --debugpy flag to enable and --debugpy-port to specify port number (default 5678)


___

### **PR Type**
Enhancement, Dependencies


___

### **Description**
- Added Debugpy support for remote Python debugging
  - New `--debugpy` and `--debugpy-port` CLI options for server
  - Server disables auto-reload when Debugpy is active
  - Debugpy port configurable (default 5678)
  - Debugpy state passed to client for developer tools

- Updated developer dependencies to include Debugpy

- Minor cleanup and improved help text for CLI options


___



### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>serverwsgi.py</strong><dd><code>Add Debugpy CLI options and server integration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

gnrpy/gnr/web/serverwsgi.py

<li>Added <code>--debugpy</code> and <code>--debugpy-port</code> CLI options<br> <li> Integrated Debugpy initialization in server run logic<br> <li> Disabled auto-reload when Debugpy is enabled<br> <li> Updated help text for debugger option<br> <li> Display Debugpy status in server startup info<br> <li> Changed reloader type to 'auto'


</details>


  </td>
  <td><a href="https://github.com/genropy/genropy/pull/171/files#diff-d3c2c6d4e7122bf9ec5cc68b4285db0e75248972f25f4f2ea141b7c568dd8735">+24/-12</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>gnrwsgisite.py</strong><dd><code>Add debugpy attribute to site and cleanup</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

gnrpy/gnr/web/gnrwsgisite.py

<li>Added <code>debugpy</code> attribute to site instance<br> <li> Removed unused error_smtp_kwargs and profiling middleware code


</details>


  </td>
  <td><a href="https://github.com/genropy/genropy/pull/171/files#diff-066ab6a11fba3cefe40e8510ad698f7840a4e9bb0916dde2205754ba6177447c">+1/-16</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>gnrwebpage.py</strong><dd><code>Pass debugpy state to client and cleanup</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

gnrpy/gnr/web/gnrwebpage.py

<li>Pass <code>debugpy</code> state to client argument dictionary<br> <li> Removed legacy error email handling code


</details>


  </td>
  <td><a href="https://github.com/genropy/genropy/pull/171/files#diff-8061bc9ceac0be38c2967721b1aba48dff9bc56d2b97c2b96d1f3a331083b669">+2/-7</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>genro.js</strong><dd><code>Add debugpy flag to GenroClient</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

gnrjs/gnr_d11/js/genro.js

- Store `debugpy` flag from server in GenroClient


</details>


  </td>
  <td><a href="https://github.com/genropy/genropy/pull/171/files#diff-722791c99c079fd5c88613da0d8092364c8d574d09e982b77fc767678da468b2">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>genro_rpc.js</strong><dd><code>Adjust XHR timeout for Debugpy debugging</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

gnrjs/gnr_d11/js/genro_rpc.js

- Set unlimited XHR timeout when Debugpy is active


</details>


  </td>
  <td><a href="https://github.com/genropy/genropy/pull/171/files#diff-801afd04e4d33484847573e25a52d2ddeb46e115ddc370c7322698b2b87160b6">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Dependencies</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>pyproject.toml</strong><dd><code>Add Debugpy to developer dependencies</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

gnrpy/pyproject.toml

<li>Added <code>debugpy</code> to developer dependencies<br> <li> Relaxed werkzeug version requirement<br> <li> Minor formatting cleanup in dependencies


</details>


  </td>
  <td><a href="https://github.com/genropy/genropy/pull/171/files#diff-d762847663d5e2a81034875a1f6b4db37b0155f5209b02c0eb8e1d8db0596b99">+3/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> <details> <summary>  Need help?</summary><li>Type <code>/help how to ...</code> in the comments thread for any questions about Qodo Merge usage.</li><li>Check out the <a href="https://qodo-merge-docs.qodo.ai/usage-guide/">documentation</a> for more information.</li></details>

mbertoldi committed:
Aggiunto supporto per Debugpy e relative opzioni di porta (default 5678) per il debug remoto

mbertoldi committed:
Cleanup

mbertoldi committed:
Merge branch 'develop' of github.com:genropy/genropy into feature/debugpy

cgabriel committed:
Merge branch 'develop' into feature/debugpy

cgabriel committed:
moved debugpy dependency to developer profile

mbertoldi committed:
Merge branch 'develop' of github.com:genropy/genropy into feature/debugpy

mbertoldi committed:
Merge branch 'develop' of github.com:genropy/genropy into feature/debugpy

fporcari committed:
Add debugpy support for server-side Python debugging

fporcari committed:
Add thbtn_batchButton method for batch operations in th_view

fporcari committed:
Merge branch 'develop' into feature/debugpy

mbertoldi committed:
Fix gnrwsgisite (debugpy attribute) initialization to handle missing options parameter (all use cases except gnr web serve)

mbertoldi committed:
Merge branch 'develop' of github.com:genropy/genropy into feature/debugpy

cgabriel commented on gnrpy/gnr/web/serverwsgi.py:
```diff
@@ -368,9 +372,15 @@ def get_instance_config(self):
+        self.debugpy_port = int(self.options.debugpy_port) if self.options.debugpy_port else 5678
+        self.reloader = not self.debugpy and not (self.options.reload == 'false' or self.options.reload == 'False' or self.options.reload == False or self.options.reload == None)
         self.debug = not (self.options.debug == 'false' or self.options.debug == 'False' or self.options.debug == False or self.options.debug == None)
-        self.start_sitedaemon()
```
Why the daemon is not being started anymore?

mbertoldi replied:
The sitedaemon option is never used and undocumented and can lead to problems in case of automatic reload. External gnrdaemon is always recomended.

cgabriel reviewed (commented):
Looks good to me, I only don't get why the start_sitedaemon has been commented out.

cgabriel committed:
Merge branch 'develop' into feature/debugpy

cgabriel committed:
the --debugpy option will only work with debugpy installed, otherwise
it will suggest to install it. Since the debugpy dependency is only
available via the genropy's developer installation profile, it will
suggest it too.

cgabriel committed:
explicitly pass the debugpy parameter to GnrWsgiSite, in order
to propagate previous checks about debugpy availability

improved argparse configuration to ensure correct type for debugpy_port,
since an int() casting is not enough.

improved logging
